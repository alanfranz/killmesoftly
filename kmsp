#!/usr/bin/env bash
source "$(dirname $0)/kms_functions"

[ -z "$1" ] && { 
    echo -e "Kill processes softly by pid number"
    echo -e "usage: $(basename "${0}") first_pid_to_kill [second_pid_to_kill] <...>\n"
    exit 1
}

while (( "$#" )); do
    [[ ! "$1" =~ ^[0-9]+$ ]] && { echo "Invalid process pid ${1}, skipping"; shift; continue; }
    checkPsExists $1 || { echo "PID ${1} not found among running processes, skipping"; shift; continue; }
    killmesoftly $1
    shift
done
