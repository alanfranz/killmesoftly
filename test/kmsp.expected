PID X not found among running processes, skipping
Now sending SIGTERM to PID X... didn't terminate, sending SIGKILL... killed successfully
Now sending SIGTERM to PID X... killed successfully
Now sending SIGTERM to PID X... killed successfully
Now sending SIGTERM to PID X... killed successfully
